MINOR: Correct Connect docs on connector/task states

C0urante opened the pull request:
The `DESTROYED` state is represented internally as a tombstone record when running in distributed mode ([1]) and by the removal of the connector/task from the in-memory status map when running in standalone mode ([2], [3]). As a result, it will never appear to users of the REST API, and we should remove mention of it from our docs so that developers creating tooling against the REST API don't write unnecessary logic to account for that state.

[1] - https://github.com/apache/kafka/blob/3dacdc5694da5db283524889d2270695defebbaa/connect/runtime/src/main/java/org/apache/kafka/connect/storage/KafkaStatusBackingStore.java#L318

[2] - https://github.com/apache/kafka/blob/3dacdc5694da5db283524889d2270695defebbaa/connect/runtime/src/main/java/org/apache/kafka/connect/storage/MemoryStatusBackingStore.java#L64-L65

[3] - https://github.com/apache/kafka/blob/3dacdc5694da5db283524889d2270695defebbaa/connect/runtime/src/main/java/org/apache/kafka/connect/storage/MemoryStatusBackingStore.java#L77-L78

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

C0urante committed:
MINOR: Correct Connect docs on connector/task states

The `DESTROYED` state is represented internally as a tombstone record when running in distributed mode ([1]) and by the removal of the connector/task from the in-memory status map when running in standalone mode ([2], [3]). As a result, it will never appear to users of the REST API, and we should remove mention of it from our docs so that developers creating tooling against the REST API don't write unnecessary logic to account for that state.

[1] - https://github.com/apache/kafka/blob/3dacdc5694da5db283524889d2270695defebbaa/connect/runtime/src/main/java/org/apache/kafka/connect/storage/KafkaStatusBackingStore.java#L318

[2] - https://github.com/apache/kafka/blob/3dacdc5694da5db283524889d2270695defebbaa/connect/runtime/src/main/java/org/apache/kafka/connect/storage/MemoryStatusBackingStore.java#L64-L65

[3] - https://github.com/apache/kafka/blob/3dacdc5694da5db283524889d2270695defebbaa/connect/runtime/src/main/java/org/apache/kafka/connect/storage/MemoryStatusBackingStore.java#L77-L78

C0urante commented:
@mimaison do you have time for this quick change?

mimaison reviewed (approved):
Thanks @C0urante for the PR!

mimaison commented:
We should also update the [Connect metrics](https://kafka.apache.org/documentation/#connect_monitoring) section that mentions `DESTROYED` as a possible state.

C0urante committed:
Remove mention of DESTROYED state from connector and task status JMX metrics

C0urante committed:
Add RESTARTING state to Connect status docs

C0urante commented:
Good point, I've removed mention of the `DESTROYED` state from that section as well.

I've also updated both to include the newly-introduced `RESTARTING` state since it was missing.

We could go further and remove the `connector-destroyed-task-count` JMX metric, but that might be overboard since (even though it's never going to be anything except 0 in practice) it is technically public interface and may require a KIP or at least a ping on the dev mailing list to get rid of.

We could possibly add a note to the docs stating that this metric is currently never updated to be anything except 0. Thoughts?
